generated/nidaqmx/_library_interpreter.py

```diff
@@ -48,10 +48,21 @@ def add_network_device(
                         ctypes_byte_str, ctypes_byte_str, c_bool32,
                         ctypes.c_double, ctypes.c_char_p, ctypes.c_uint]
 
-        error_code = cfunc(
-            ip_address, device_name, attempt_reservation, timeout,
-            ctypes.byref(device_name_out))
-        check_for_error(error_code)
+        temp_size = 0
+        while True:
+            device_name_out = ctypes.create_string_buffer(temp_size)
+            size_or_code = cfunc(
+                ip_address, device_name, attempt_reservation, timeout,
+                device_name_out, temp_size)
+            if is_string_buffer_too_small(size_or_code):
+                # Buffer size must have changed between calls; check again.
+                temp_size = 0
+            elif size_or_code > 0 and temp_size == 0:
+                # Buffer size obtained, use to retrieve data.
+                temp_size = size_or_code
+            else:
+                break
+        check_for_error(size_or_code)
         return device_name_out
```

Comment:
This returns a raw ctypes string buffer. It should convert to a string: `return device_name_out.value.encode('ascii')`